mozilla/fxa
# feat(react): Activate CompleteSignin page in React

## Because

* We need to keep this page active for backwards compatibility with Firefox version <122

## This pull request

* Conditionally route traffic for `complete_signin` to React
* Add `complete_signin` to React routes
* Enable signin verification with React
* Add unit tests and stories
* Fix `mockSession` for use in storybook

## Issue that this pull request solves

Closes: #FXA-6483, FXA-9073

## Checklist

_Put an `x` in the boxes that apply_


## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Testing instructions:
1. Force signin confirmation by setting the `fxa-auth-server/config/index.ts` default for `forceGlobally` to `true` (optionally/alternatively, can also set `skipForNewAccounts` to `false` and `maxAge` to 0)
2. _(may need to restart the stack)_
3. Download and install an older (<122) version of Firefox - for example, ESR is currently on v115
4. Use fxa-dev-launcher to start the older instance of firefox with local server configuration. On Mac, this could look like:
`FIREFOX_BIN=/Applications/FirefoxESR.app/Contents/MacOS/firefox-bin npm start`
5. Create an account from localhost:3030, then sign in to Sync from the hamburger menu. DO NOT ENTER A CONFIRMATION CODE
6. In the Firefox hamburger menu, click "Finish account setup" then "Resend verification"
7. COPY the link received by email and append the react param `&showReactApp=true`
8. Paste the link in the same browser or a different browser (should test both) and observe if sign in to sync on original browser is successful

To be completed in other tickets:
- Adding metrics: `FXA-9132`
- Adding functional test if needed : `FXA-9133`
